subprojects/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/EclipseClasspath.java

```diff
@@ -145,9 +148,12 @@
 
     private final org.gradle.api.Project project;
 
+    private final ModularClasspathHandling modularClasspathHandling;
```

Comment:
❌ Why do we need this here? As far as I can tell there should not be any Eclipse-specific configuration with regards to the module system. I'd try to remove this from the `eclipse` plugin's public API.

Reply:
You are right. I did this in the beginning when I didn't have the full picture and then I did not question it again. I removed the API and now look up the value in the Java compile task the moment we resolve the classpath for Eclipse.

It's also a bit more correct this way. If we look at the task, we will also get the right value if it is only set for the task (and not on the extension).